src/types/Health.ts

```diff
@@ -354,10 +362,59 @@ export class WorkloadHealth extends Health {
   }
 }
 
+export const calculateErrorRatio = (health: RequestHealth): RequestHealthRatio => {
+  var req: RequestHealthRatio = { errorRatio: -1, inboundErrorRatio: -1, outboundErrorRatio: -1 };
+
+  const inbound = getErrorRatio(health.inboundRequestsCode);
+  req.inboundErrorRatio = inbound.errorRatio;
+
+  const outbound = getErrorRatio(health.outboundRequestsCode);
+  req.outboundErrorRatio = outbound.errorRatio;
+
+  const globalRequestRate = inbound.reqRate + outbound.reqRate;
+  const globalErrorRate = inbound.errorRate + outbound.errorRate;
+
+  if (globalRequestRate == 0) {
+    req.errorRatio = -1;
+  } else {
+    req.errorRatio = globalErrorRate / globalRequestRate;
+  }
+  return req;
+};
+
+type errorsRate = { errorRatio: number; errorRate: number; reqRate: number };
+
+const regexProtocol = {
+  grpc: new RegExp('^[1-9]$|^1[0-6]$'),
+  http: new RegExp('^[4-5]dd$')
```

Comment:
I guess some backslashes are missing here: `[4-5]\d\d`

Reply:
Mmm copy paste it remove them hahah